lib/ClosureF.hs

```diff
@@ -67,12 +67,11 @@ data Expr t e =
    | JNew ClassName [Expr t e]
    | JMethod (Either ClassName (Expr t e)) MethodName [Expr t e] ClassName
    | JField  (Either ClassName (Expr t e)) FieldName ClassName
-   | PolyList [Expr t e] (Type t)
-   | JProxyCall (Expr t e) (Type t)
    | SeqExprs [Expr t e]
    | Data S.RecFlag [DataBind t] (Expr t e)
    | Constr (Constructor t) [Expr t e]
    | Case (Expr t e) [Alt t e]
+   | Error (Type t) (Expr t e)
```

Comment:
this is for throwing error in program. The type is for indicating the return type. The expr is a string for error message. like

```
head (x:List[T]) : T = 
  case x of
     Cons y _ -> y
  |  Nil -> error [T] "head of empty"
```